nixos/modules/services/databases/mysql.nix

```diff
@@ -482,6 +479,35 @@ in
               # ensureDatbases & ensureUsers depends on this script being run as root
               # when the user has secured their mysql install
               "+${setupScript}";
+          # User and group
+          User = cfg.user;
+          Group = "mysql";
+          # Runtime directory and mode
+          RuntimeDirectory = "mysqld";
+          RuntimeDirectoryMode = "0755";
+          # Access write directories
+          ReadWritePaths = [ cfg.dataDir ];
```

Comment:
With the service running as a non `root` user, `NoNewPrivileges` `RestrictSUIDSGID`, `ProtectHome`, `PrivateTmp`, and `PrivateMounts` all set I don't see the value of `ReadWritePaths` here. Seems to just cause hassle to the user for a very normal process: `LOAD DATA INFILE`. I would suggest removing this option. I think all the other options are pretty good, though.

I'd like to test this out.

Reply:
This option need to use custom location `dataDir`.
Example:
```
dataDir = "/var/data/mysql";
```

Reply:
What option is causing the service to have no access to `/var/data/mysql` if `/var/data/mysql` is owned by `cfg.user:mysql`?

Reply:
In mode `ProtectSystem = "strict";` , by default, read and write access is only available to the folder `/run/mysql/`

Reply:
Ah, right. Maybe we should drop that one then. The `cfg.user` user only has write access to `cfg.dataDir` anyways, unless the user has specified otherwise.

Reply:
Better to completely restrict access.

Reply:
Since the `cfg.user` user isn't `root` my opinion is that `ProtectSystem = "full";` is sufficient. If the `cfg.user` has rights on any other path it is because the sysadmin has explicitly granted it already.

Reply:
Strict mode is safer.
To work, the user does not need to change anything in his configuration.

`If set to "strict" the entire file system hierarchy is mounted read-only, except for the API file system subtrees /dev, /proc and /sys (protect these directories using PrivateDevices=, ProtectKernelTunables=, ProtectControlGroups=). This setting ensures that any modification of the vendor-supplied operating system (and optionally its configuration, and local mounts) is prohibited for the service. It is recommended to enable this setting for all long-running services, unless they are involved with system updates or need to modify the operating system in other ways. If this option is used, ReadWritePaths= may be used to exclude specific directories from being made read-only.`

Reply:
I agree with @Izorkin it protects against miss-configuration there not many good reasons for `mysql` to read home.

Reply:
This is what `ProtectHome` is for.

Reply:
You both make good arguments. After carefully reviewing `mysql` and `systemd` documentation I now also think this is a good idea :+1: